### PR TITLE
refactor!: separate expressions from execution plans && renames

### PR DESCRIPTION
**Pull request by iajoiner**
# Rationale for this change
As the amount of expressions and execution plans increase we need to seriously differentiate between the two. This PR is the first in a series of PRs that will lead to `SliceExec` and `UnionExec`.
 

# What changes are included in this PR?
- rename `ProvableExpr` to `ProofExpr`
- rename `ProvableExprPlan` to `DynProofExpr`
- rename `AliasedProvableExprPlan` to `AliasedDynProofExpr`
- rename `ProvableExprPlanBuilder` to `DynProofExprBuilder`
- rename `ProofPlan` to `DynProofPlan`
- rename `ProofExecutionPlan` to `ProofPlan`
- move all `ProofExpr`s to `sql::proof_exprs`
- move all `ProofPlan`s to `sql::proof_plans`
- fix `Cargo.toml` warning in new Rust
 

# Are these changes tested?
Existing tests should pass

### CODE REVIEW

**Review by JayWhite2357 — COMMENTED** [review]
Adding another level of nesting is a bit too much. Three levels is plenty.
I would prefer for this to be ~`sql::physical_expr`~`sql::provable_expr` and ~`sql::physical_plan`~`sql::provable_plan`.

Additionally, I prefer `ProvableExpr` over `PhysicalExpr` because it is more clear, and prefer it over `ProvablePhysicalExpr` because we won't have any other kinds of "provable expressions", so it's a needless disambiguation that is not concise.

**Comment by JayWhite2357** [comment]
Also, can you structure the commits a bit more so that each change is separate and easily reviewable?
i.e. do a type rename in a single commit and the file renames/moves in a separate commit.

And if something shows up as a delete/create rather than a move, it would be nice for that to be isolated to it's own commit so that it's easy to understand.

I'll review each commit in isolation, so as long as each of them makes sense and is manageable, it won't matter how huge the PR grows.

**Comment by JayWhite2357** [comment]
A final suggestion while we're renaming: alloy uses `DynSolType` to refer to an enum that is "Equivalent to an enum wrapper around all implementers of `SolType`."

So, we could name our expression trait `ProvableExpr`, and the enum `DynProvableExpr`. Then, name our plan trait `ProvablePlan`, and the enum `DynProvablePlan`. With this, I think it would be fairly clear and obvious what each thing is.

**Comment by iajoiner** [comment]
Really thanks for the review! I will basically redo this.

File: crates/proof-of-sql/src/sql/proof_expr/mod.rs [hunk 1]
```diff
@@ -81,3 +79,57 @@ mod column_expr;
...
+        proof::ProofError,
+    },
+    sql::proof::{CountBuilder, ProofBuilder, VerificationBuilder},
+};
+use bumpalo::Bump;
+use indexmap::IndexSet;
+use std::fmt::Debug;
+
+/// Provable AST column expression that evaluates to a `Column`
+pub trait ProofExpr<C: Commitment>: Debug + Send + Sync {
```
[the remark this answers is not in the document]
Comment (iajoiner) [reply crates/proof-of-sql/src/sql/proof_expr/mod.rs:96]: I picked option 1

**Review by JayWhite2357 — APPROVED** [review]

**Pull request merged**